projects/sandbox/data_generation/data_generation/utils.py

```diff
@@ -4,10 +4,27 @@
 import numpy as np
 import torch
 from gwpy.timeseries import TimeSeries, TimeSeriesDict
+from pycbc.noise import noise_from_psd
```

Comment:
I completely agree, I was thinking the same thing. I did this out of laziness as `lalsimulation` is really not a friendly api, especially since you don't get descriptive error messages. I was trying to implement that exact function you linked and was running into problems. Can I suggest we open this as an issue for now?

Reply:
How about something adapted from the following
```
import lal, lalsimulation
import matplotlib.pyplot as plt

flow = 40.0
duration = 16.0
srate = 16384.0
length = int(duration * srate)
stride = length // 2

epoch = lal.LIGOTimeGPS(0, 0)
rng = lal.gsl_rng("ranlux", 0)

seg = lal.CreateREAL8TimeSeries("STRAIN", epoch, 0.0, 1.0/srate, lal.StrainUnit, length)
psd = lal.CreateREAL8FrequencySeries("LIGO SRD", epoch, 0.0, 1.0/duration, lal.SecondUnit, length//2 + 1)

lalsimulation.SimNoisePSDFromFile(psd, flow, './AplusDesign.txt')
lalsimulation.SimNoise(seg, 0, psd, rng)

times = [seg.deltaT * ii for ii in range(seg.data.length)]
plt.plot(times, seg.data.data)
plt.show()
```
![image](https://user-images.githubusercontent.com/23223278/223026221-27814984-7aee-4157-b0d6-abcb2152bc2e.png)

I took it from the docs here: https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalsimulation/lib/LALSimNoise.c#L104-118. For longer lengths, the above function with stride = 1, 2,... until desired length is achieved. The noise curve file is taken from here: https://dcc.ligo.org/LIGO-T2000012/public

Reply:
Thanks for doing this Deep, will implement and get rid of pycbc.